modules/gdscript/gdscript_utility_functions.cpp

```diff
@@ -144,21 +144,21 @@ struct GDScriptUtilityFunctionsDefinitions {
 			} break;
 			case 1: {
 				VALIDATE_ARG_NUM(0);
-				int count = *p_args[0];
-				Array arr;
-				if (count <= 0) {
+				TypedArray<int64_t> arr;
+				int64_t count = *p_args[0];
+				if (count <= 0 || count > (int64_t)std::numeric_limits<int>::max()) {
```

Comment:
It's `int`'s max which is needed here as `count` (input) is `int64_t` and it's passed to `Array::resize` which takes `int`. So this check is to avoid truncation. I could change it to `(int64_t)INT32_MAX` instead but not sure if it's portable enough ([`int` can be 16-bit or 64-bit as well](https://en.cppreference.com/w/cpp/language/types)). Not sure what are the assumptions for `int`s in the Godot's codebase. :thinking: